File: calendar-reminders-backend/src/main/java/lt/calendar/reminders/domain/reminder/Reminder.java
package lt.calendar.reminders.domain.reminder;

import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class Reminder {

    @EqualsAndHashCode.Include
    private Long id;
    private String text;
    private LocalDate date;
    private LocalTime time;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/src/types/reminder.ts
export interface Reminder {
    id: number;
    text: string;
    date: string;
    time: string;
    createdAt: string;
    updatedAt: string;
}
export type ReminderPayload = Omit<Reminder, "id" | "createdAt" | "updatedAt">;